src/Serilog.Sinks.AzureDataExplorer/AzureDataExplorerSinkOptions.cs

```diff
@@ -78,30 +120,43 @@ public int QueueSizeLimit
         public string ApplicationKey { get; private set; }
         public X509Certificate2 ApplicationCertificate { get; private set; }
         public string Authority { get; private set; }
-        public bool SendX5c { get; private set; }
+        public bool SendX5C { get; private set; }
         public string AzureRegion { get; private set; }
         public Azure.Core.TokenCredential TokenCredential { get; private set; }
 
         public AzureDataExplorerSinkOptions()
         {
-            Period = TimeSpan.FromSeconds(10);
-            BatchPostingLimit = 1000;
-            QueueSizeLimit = 100000;
+            this.Period = TimeSpan.FromSeconds(10);
+            this.BatchPostingLimit = 1000;
+            this.QueueSizeLimit = 100000;
+            this.BufferFileRollingInterval = RollingInterval.Hour;
+            this.BufferFileCountLimit = 31;
+            this.BufferFileSizeLimitBytes = 100L * 1024 * 1024;
+            this.BufferFileOutputFormat =
+                "{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level:u3}] {Message:lj}{NewLine}{Exception}";
+            this.FlushImmediately = false;
         }
 
         #region Authentication builder methods
-        public AzureDataExplorerSinkOptions WithAadApplicationCertificate(string applicationClientId, X509Certificate2 applicationCertificate, string authority, bool sendX5c = false, string azureRegion = null)
+        public AzureDataExplorerSinkOptions WithAadApplicationCertificate(string applicationClientId, X509Certificate2 applicationCertificate, string authority, bool sendX5C = false, string azureRegion = null)
         {
             AuthenticationMode = AuthenticationMode.AadApplicationCertificate;
             ApplicationClientId = applicationClientId;
             ApplicationCertificate = applicationCertificate;
             Authority = authority;
-            SendX5c = sendX5c;
+            SendX5C = sendX5C;
             AzureRegion = azureRegion;
 
             return this;
         }
 
+        public AzureDataExplorerSinkOptions WithAadManagedIdentity(string applicationClientId)
+        {
```

Comment:
precautionary comment: ManagedIdentity could be of two types, System assigned (without clientId) and User assigned (with clientId), hopefully we are covering both.